packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -30,3 +32,121 @@ export function logIfFalse(
 	logger.send(newEvent);
 	return false;
 }
+
+/**
+ * Wraps around an existing logger matching the {@link ITelemetryBaseLogger} interface and applies
+ * a provided callback to determine if an event should be sampled in conjunction with the 'Fluid.Telemetry.DisableSampling' logger config value.
+ * If Fluid.Telemetry.DisableSampling is set to true, all events will be unsampled, otherwise they will be sampled according to your provided event sampler callback.
+ */
+export function createSampledLogger(
+	logger: ITelemetryBaseLogger,
+	eventSampler: {
+		poll: () => boolean;
+	},
+) {
+	const monitoringContext = loggerToMonitoringContext(logger);
+	const isSamplingEnabled = !monitoringContext.config.getBoolean(
+		"Fluid.Telemetry.DisableSampling",
+	);
+
+	const sampledLogger: ITelemetryBaseLogger = {
+		send: (event: ITelemetryBaseEvent) => {
+			if (isSamplingEnabled && eventSampler.poll() === true) {
+				logger.send(event);
+			}
+		},
+	};
+
+	return sampledLogger;
+}
+/**
+ * Wraps around an existing logger matching the {@link ITelemetryLoggerExt} interface and applies
+ * a provided callback to determine if an event should be sampled in conjunction with the 'Fluid.Telemetry.DisableSampling' logger config value.
+ * If Fluid.Telemetry.DisableSampling is set to true, all events will be unsampled, otherwise they will be sampled according to your provided event sampler callback.
+ */
+export function createSampledLoggerExt(
```

Comment:
Since this is the only one we export out of the package, and to match the terminology we use in other places, I'd say let's call this `createSampledLogger()` and the other one `createSampledBaseLogger()`. Also worth adding a `@remarks` to this one saying that we specifically wrap `ITelemetryLoggerExt` so we can avoid any work that might be done by its public API members (vs if we wrapped `ITelemetryBaseLogger`, the wrapped logger might still do some processing before calling `send()` and realize that the message wasn't necessary).